### PR TITLE
Feature/msm test

### PR DESCRIPTION
**Pull request by ulianaStashkiv**

### CODE REVIEW

File: applications/MachineStateManager/test/state_machine_test.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    void TearDown() override
+    {
+        delete msmMock;
+    }
+    
+   std::unique_ptr<StateClientMock> stateClientMock{nullptr};
+   std::unique_ptr<IStateFactoryMock> factoryMock{nullptr};
+   IMachineStateManagerMock* msmMock{nullptr};
```
Comment (RomanLevkovych) [inline applications/MachineStateManager/test/state_machine_test.cpp]: Do we need this pointer to be raw?

Comment (RomanLevkovych) [reply applications/MachineStateManager/test/state_machine_test.cpp]: Btw, in tests above, you init state with this parameter. Let's replace this tests with one and replace `TEST_F` with `TEST_P` and prepare it in respective `SetUp`

File: applications/MachineStateManager/test/state_machine_test.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    state->enter();
+}
+
+TEST_F(MsmStateMachineTest, Should_TerminateCallLeave)
+{
+    std::unique_ptr<ShutDown> state = std::make_unique<ShutDown>(*msmMock);
+    state->leave();
+}
+
+TEST_F(MsmStateMachineTest, Should_CreateInit)
```
Comment (RomanLevkovych) [inline applications/MachineStateManager/test/state_machine_test.cpp]: Tests duplicates, we can replace them with one too.

File: applications/MachineStateManager/test/msm_test.cpp [hunk 3]
```diff
@@ -1,3 +1,92 @@
...
+{
+protected:
+   std::unique_ptr<StateClientMock> stateClientMock{std::make_unique<StateClientMock>()};
+   std::unique_ptr<IStateFactoryMock> factoryMock{std::make_unique<IStateFactoryMock>()};
+   std::unique_ptr<NiceMock<IStateMock>> stateInitMock = std::make_unique<NiceMock<IStateMock>>();
+   std::unique_ptr<NiceMock<IStateMock>> stateRunMock = std::make_unique<NiceMock<IStateMock>>();
+   std::unique_ptr<NiceMock<IStateMock>> stateTermMock = std::make_unique<NiceMock<IStateMock>>();
+};
+
+TEST_F(MsmTest, Should_TransitToInitState)
```
Comment (RomanLevkovych) [inline applications/MachineStateManager/test/msm_test.cpp]: Test cases and suites names shouldn't contain *_*

File: applications/MachineStateManager/test/msm_test.cpp [hunk 4]
```diff
@@ -1,3 +1,92 @@
...
+            .WillOnce(Return(ByMove(std::move(stateInitMock))));
+
+     MachineStateManager::MachineStateManager msm{std::move(factoryMock), nullptr};
+     msm.init();
+
+}
+
+TEST_F(MsmTest, Should_TransitToRunState)
+{
+    EXPECT_CALL(*factoryMock,
```
Comment (RomanLevkovych) [inline applications/MachineStateManager/test/msm_test.cpp]: Let's consider moving this `EXPECT_CALL` to SetUp.  Drawback is failing of `Should_ReportCurrentState`

File: applications/MachineStateManager/include/msm_state_machine.hpp [hunk 5]
```diff
@@ -5,13 +5,13 @@
...
 
-namespace MachineStateManager
+namespace MSM
 {
 
 class MsmState : public api::IState
 {
 public:
-    MsmState(MachineStateManager& msm, 
+    explicit MsmState(MachineStateManager& msm,
```
Comment (rBekesh) [inline applications/MachineStateManager/include/msm_state_machine.hpp]: explicit not needed

**Pull request merged**